File: app/src/main/java/com/example/weatherapp/WeatherDatabase.java
// WeatherDatabase.java
package com.example.weatherapp;

import android.content.Context;
import androidx.room.Database;
import androidx.room.Room;
import androidx.room.RoomDatabase;

@Database(entities = {City.class}, version = 1, exportSchema = false)
public abstract class WeatherDatabase extends RoomDatabase {
    private static volatile WeatherDatabase INSTANCE;
    public abstract CityDao cityDao();

    public static WeatherDatabase getInstance(Context ctx) {
        if (INSTANCE == null) {
            synchronized (WeatherDatabase.class) {
                if (INSTANCE == null) {
                    INSTANCE = Room.databaseBuilder(ctx.getApplicationContext(), WeatherDatabase.class, "weather_db")
                            .fallbackToDestructiveMigration().build();
                }
            }
        }
        return INSTANCE;
    }
}

File: app/src/main/java/com/example/weatherapp/WeatherDatabase.kt
package com.example.weatherapp

import androidx.room.Database
import androidx.room.Room
import androidx.room.RoomDatabase

@Database(
    entities = [LocationEntity::class, ForecastDayEntity::class],
    version = 1,
    exportSchema = false
)
abstract class WeatherDatabase : RoomDatabase() {
    abstract fun locationDao(): LocationDao
    abstract fun forecastDao(): ForecastDao

    companion object {
        val instance: WeatherDatabase by lazy {
            Room.databaseBuilder(
                WeatherApp.instance,
                WeatherDatabase::class.java,
                "weather.db"
            )
                .fallbackToDestructiveMigration()
                .build()
        }
    }
}
